src/html-pipe.js

```diff
@@ -82,13 +82,19 @@ export async function htmlPipe(state, req) {
     }
 
     // ...and apply the folder mapping
-    await folderMapping(state);
+    state.timer?.update('content-fetch');
+    let contentPromise = await fetchContent(state, req, res);
+    // ...but only if the current resource doesn't exist
+    if (res.status === 404) {
+      await folderMapping(state);
+      contentPromise = fetchContent(state, req, res);
```

Comment:
this only needs to be executed if the `state.info.mapped === true`, i.e. if there was an actual remapping.
otherwise the same missing content is fetched twice.